app/models/facet_config.rb

```diff
@@ -0,0 +1,48 @@
+class FacetConfig < ApplicationRecord
+  belongs_to :metadata_coverage
+  after_commit :populate_facets, on: :update
+  validates :order, numericality: true, allow_blank: true
+  validates :order, inclusion: { in: 0..9, if: :type_text? }, allow_blank: true
+  validates :order, inclusion: { in: 0..2, if: :type_date? }, allow_blank: true
+
+  def populate_facets
+    works = self.metadata_coverage.collection.works
+
+    works.each do |w|
+      unless w.original_metadata.nil?
+        om = JSON.parse(w.original_metadata)
+
+        unless w.work_facet
+          w.create_work_facet
+        end
+
+        om.each do |o|
+          label = o['label']
+
+          if label == self['label']
```

Comment:
I just realized that the `label` field which users can enter on `facet_config` refers to the label we plan to display to users searching the collection, not the label in the work's `original_metadata` that lets us look up fields. 

We have two separate ideas of "labels" on `facet_config`:
* `label` is a text string entered by the user configuring facets, and is stored within `facet_config` but *is not used to look up values in `work.original_metadata`*
* `original_label` is a text string from `metadata_coverage` which the user cannot edit.  It (currently) lives in `metadata_coverage.key` and it *should be used to look up values in `work.original_metadata`.

I think that changing line 22 to be `if label == self.metadata_coverage.key` will solve this issue.